src/hopen/syntax/mustache.cljc

```diff
@@ -0,0 +1,434 @@
+  ast)
+
+(defn ast-node->hopen-node
+  [node]
+
+  (cond
+    (string? node) node
+    (map? node) (cond
+                  (= :section-open (:type node)) node
+                  (= :var-ref (:type node))
+                  (if (:skip-escape node)
+                    (backtick/template
+                     (ctx-lookup ~(keyword (:var-name node))))
+                    (backtick/template
+                     (escape-html (ctx-lookup ~(keyword (:var-name node))))))
+                  (= :section-close (:type node)) node
+                  (= :inverted-open (:type node)) node
+                  (= :comment (:type node)) nil
+                  (= :partial (:type node)) nil
+                  (= :delim-change (:type node)) nil
+                  :else (throw (ex-info (str "Does not how how to deal with tags of type: " (:type node)) {})))
+
+    (vector? node)
+    (cond
+      (= :section-open (:type (first node))) (let [contents (->> node
+                                                                 (drop 1)
+                                                                 (drop-last 1)
+                                                                 (into []))]
+                                               (backtick/template
+                                                (b/let [data (ctx-lookup ~(keyword (:section-name (first node))))]
+                                                  [(b/if (and data
+                                                              (not (empty? data)))
+                                                     [(b/if (or (list? data)
+                                                                (vector? data))
+
+                                                        ;; Looks like the context might have multiple items
+                                                        [(b/for [ctx data]
+                                                           [(b/let [mustache/ctx (conj mustache/ctx ctx)]
+                                                              ~contents)])]
+
+                                                        ;; New context only has a single item
+                                                        [(b/let [mustache/ctx (conj mustache/ctx data)]
+                                                           ~contents)])])])))
+
+      (= :inverted-open (:type (first node))) (backtick/template
+                                               (b/let [data (ctx-lookup ~(keyword (:section-name (first node))))]
+                                                 [(b/if (or (= data nil)
+                                                            (empty? data))
+                                                    ~(->> node
+                                                          (drop 1)
+                                                          (drop-last 1)
+                                                          (into [])))])))))
+
+(defn postwalk-zipper [f loc opts]
```

Comment:
I suggest the parameter order `(defn postwalk-zipper [loc f opts] ...)` instead, so that you can chain the zipper functions using `->`.